hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/SplitTableRegionProcedure.java

```diff
@@ -854,7 +854,8 @@ private ServerName getParentRegionServerName(final MasterProcedureEnv env) {
     for (int i = 0; i < procs.length; ++i) {
       final RegionInfo hri = RegionReplicaUtil.getRegionInfoForReplica(getParentRegion(), i);
       procs[i] = env.getAssignmentManager().
-          createUnassignProcedure(hri, null, true, !RegionReplicaUtil.isDefaultReplica(hri));
+          createUnassignProcedure(hri, null, true,
+            !RegionReplicaUtil.isDefaultReplica(hri));
```

Comment:
@sreenivasulureddy , the original fix that I had reverted, as explained on the PR description.

Reply:
@waterlx ,Ok fine for this revert.

Reply:
@wchevreuil , ok fine for this revert

Reply:
The above solution i have given based on, while disabling the table, both primary region and  Replica region  handling in common way.